core/src/test/java/org/elasticsearch/search/aggregations/metrics/percentiles/InternalPercentilesRanksTestCase.java

```diff
@@ -64,5 +75,36 @@ protected final void assertFromXContent(T aggregation, ParsedAggregation parsedA
         assertTrue(parsedClass.isInstance(parsedAggregation));
     }
 
+    public void testPercentilesRanksIterators() throws IOException {
```

Comment:
maybe some code could be shared, or maybe shuffling could be disabled optionally?